### DISCUSSION TITLE
Add has_property() and has_static_property() methods

### DISCUSSION THREAD

**Discussion by azur-wolve** [godotengine/godot-proposals#10031] [Scripting]
Object has `has_method()` to easily check if the method name passed is there.
It works for both instance methods and static methods.
(Still feel that should have a `has_static_method()`)

But for properties must do a lot of boilerplate to check if has a specific property by its name. → `get_property_list()` and then do lookup on dictionaries.
For certain scenarios its fine, but when only want to check fast a single one isn't ideal.

Have coded my own utility methods for that:
```gdscript
static func has_property(obj : Object, name: String) -> bool:
	var properties: Array[Dictionary]
	
	if obj and not name.is_empty():
		properties = obj.get_property_list()
		
		for property: Dictionary in properties:s
			if property["name"] == name:
				return true
	
	return false

static func has_static_property(obj: Object, name: String):# -> bool:	
	if obj and not name.is_empty():
		var source_code: String = normalize_spaces_and_tabs(obj.get_script().source_code)
		var lines = source_code.split("\n")
		
		for line in lines:
			if line.strip_edges().begins_with("static var " + name):
				return true
	
	return false

static func normalize_spaces_and_tabs(text: String, strip_edges: bool = false) -> String: 
        #INFO: replaces consecutive spaces and tabs by single spaces
	# complements String.strip_edges() which only strips at beginning and end
	var regex = RegEx.new()
	regex.compile("[ \t]+")
	var all := true
	var new_string: String = regex.sub(text, " ", all)
	if strip_edges:
		new_string = new_string.strip_edges()
	return new_string
```

Solution → add official methods that do that:
`Object.has_property(name: String)`

`Object.has_static_property(name: String)`

and an extra also:

`@GlobalScope.normalize_spaces_and_tabs(text: String, strip_edges: bool = false)`

In my code example is used to ensure that the source code of the script has only single spaces before checking if has `"static var " + name`.

**Comment by AThousandShips** [comment]
Thank you for your proposal, consolidating in:
* https://github.com/godotengine/godot-proposals/issues/717
